custom/ilsgateway/api.py

```diff
@@ -250,18 +245,12 @@ def location_sync(self, ilsgateway_location, fetch_groups=False):
                 location.longitude = float(ilsgateway_location.longitude)
             location.location_type = ilsgateway_location.type
             location.site_code = ilsgateway_location.code
-            # todo: unicode?
-            location.external_id = str(ilsgateway_location.id)
+            location.external_id = unicode(ilsgateway_location.id)
             location.save()
 
-            # todo: shouldn't this only be creating supply points for objects just at the facility level?
-            # explanation: There are some sms users in ILS that are assigned to non-facility locations.
-            # In HQ when we assign location to user supply point is automatically created.
-            # That's reason why I'm creating supply point for all locations. Not sure how it should be solved.
-            # note: I think we can now assign users to locations without making a supply point so am
-            # hoping this can get changed.
-            if not SupplyPointCase.get_by_location(location):
+            if ilsgateway_location.type == 'FACILITY' and not SupplyPointCase.get_by_location(location):
                 SupplyPointCase.create_from_location(self.domain, location)
+                location.save()
```

Comment:
@czue create_from_location doesn't update supply_point_id field in SQLLocation table so that's why I need this additional save. Could we modify create_from_location function to update SQLLocation table?